### PR TITLE
Rework "Accept" header handling for HTTP 500 response body on unhandled exceptions

### LINKED ISSUE #20198: Vert.x-http: response body of HTTP status 500 should be human-readable by default
### Description

Currently, when a critical (unhandled) error occurs in a REST endpoint, we generate a default response with HTTP status 500 and an HTML webpage describing the error. This is done in `QuarkusErrorHandler`.

While I doubt this matters in production, it can be really annoying when developing, as you perform an HTTP request through `curl`, and you get dozens of lines of HTML in response, right in your terminal.

If your request includes an `Accept: application/json` header, you will get JSON in the response; however, by default, you will get HTML.

### Implementation ideas

**At least in test/dev mode**, we could change the default content type of the response. When an error happens:

* If there is an `Accept` header mentioning `text/html`, `application/xhtml+xml` or `application/xml`, we return HTML.
* If there is an `Accept` header mentioning `application/json`, we return JSON.
* If there isn't any `Accept` header, or if it doesn't mention any supported content-type, we default to some human-readable format. Maybe `application/json`, or maybe just `text/plain`.

My reasoning: all clients that could potentially render HTML (browsers) will probably send an `Accept: text/html` header. 
So if we don't find one, it's probably a human directly requesting something. Or a badly configured REST client, but in that case getting JSON or plain text won't be much worse to get plain text than HTML.

### PR DESCRIPTION
**Pull request by yrodiere**
Fixes #20198

### CODE REVIEW

File: extensions/vertx-http/runtime/src/main/java/io/quarkus/vertx/http/runtime/QuarkusErrorHandler.java [hunk 1]
```diff
@@ -118,28 +134,49 @@ public void accept(Throwable throwable) {
...
+                jsonResponse(event, ContentTypes.APPLICATION_JSON, details, stack);
+                break;
         }
     }
 
+    private void jsonResponse(RoutingContext event, String contentType, String details, String stack) {
+        event.response().headers().set(HttpHeaderNames.CONTENT_TYPE, contentType + "; charset=utf-8");
+        String escapedDetails = escapeJsonString(details);
+        String escapedStack = escapeJsonString(stack);
+        StringBuilder jsonPayload = new StringBuilder("{\"details\":\"")
```
Comment (cescoffier) [inline extensions/vertx-http/runtime/src/main/java/io/quarkus/vertx/http/runtime/QuarkusErrorHandler.java:163]: Unrelated to this PR - I'm wondering if at some point we shouldn't use something like https://github.com/zalando/problem.

File: extensions/vertx-http/runtime/src/main/java/io/quarkus/vertx/http/runtime/QuarkusErrorHandler.java [hunk 2]
```diff
@@ -200,4 +237,45 @@ static String escapeJsonString(final String text) {
...
+        private static final String APPLICATION_JSON = "application/json";
+        private static final String TEXT_JSON = "text/json";
+        private static final String TEXT_HTML = "text/html";
+        private static final String APPLICATION_XML = "application/xml";
+        private static final String TEXT_XML = "text/xml";
+        private static final String APPLICATION_XHTML = "application/xhtml+xml";
+        private static final Set<String> SUPPORTED_SET = Set.of(APPLICATION_JSON, TEXT_JSON, TEXT_HTML, APPLICATION_XML,
+                TEXT_XML, APPLICATION_XHTML);
+
+        static String pickFirstSupportedAndAcceptedContentType(HttpServerRequest request) {
```
Comment (cescoffier) [inline extensions/vertx-http/runtime/src/main/java/io/quarkus/vertx/http/runtime/QuarkusErrorHandler.java:254]: I'm wondering if Vert.x does not have this code somewhere.

Comment (cescoffier) [reply extensions/vertx-http/runtime/src/main/java/io/quarkus/vertx/http/runtime/QuarkusErrorHandler.java:254]: Should we reuse: https://github.com/vert-x3/vertx-web/blob/36bf548a7362336a9838e8260c40b543fc032cc6/vertx-web/src/main/java/io/vertx/ext/web/impl/HeaderParser.java ?

**Review by cescoffier — CHANGES_REQUESTED** [review]

File: extensions/vertx-http/runtime/src/main/java/io/quarkus/vertx/http/runtime/QuarkusErrorHandler.java (hunk 2 above)
Comment (yrodiere) [reply extensions/vertx-http/runtime/src/main/java/io/quarkus/vertx/http/runtime/QuarkusErrorHandler.java:254]: Thanks! I was looking for something like that, but didn't find it. I'll change the code to use it.

**Comment by yrodiere** [comment]
@cescoffier Thanks for the review. I addressed your comment, and also incorporated #20251, along with an extra commit to change the setting from an override to just a default (see [my comment on that PR](https://github.com/quarkusio/quarkus/pull/20251#discussion_r711906466)).

**Review by cescoffier — APPROVED** [review]
LGTM! Thanks for integrating the _conflicting_ PR.

**Pull request merged**